### ISSUE TITLE
v4: PngWriter creates blurry shapes

### ISSUE THREAD

**Issue by sprain** [endroid/qr-code#355]
I noticed with v4 that qr codes written as png become blurry while in v3 the same setup created razor sharp images. See the following two examples and zoom into the qr codes to see the difference. The difference is also clearly detectable in file size.

I experimented with different RoundBlockSizeModes which did not change this effect. I also tried different enviroments (e.g. PHP 7.3 vs PHP 8) with no change to this behaviour.

**v3.9.7**
```php
$data = "AAAABBBBCCCCDDDD";

$qrCode = new QrCode($data);
$qrCode->setSize(300);
$qrCode->setMargin(0); 
$qrCode->setEncoding('UTF-8');
$qrCode->setWriterByName('png');
    
$qrCode->writeFile(__DIR__.'/qrcode.png');
```

![qrcode](https://user-images.githubusercontent.com/260361/139735125-ea180364-5734-4f1b-8824-f24c47916d32.png)

---

**v4.3.5**
```php
$data = "AAAABBBBCCCCDDDD";

$result = Builder::create()
    ->data($data)
    ->size(300)
    ->margin(0)
    ->encoding(new Encoding('UTF-8'))
    ->writer(new PngWriter())
    ->build();
    
$result->saveToFile(__DIR__.'/qrcode.png');
```
![qrcode](https://user-images.githubusercontent.com/260361/139735144-f5e63d1d-9db4-4bf4-8ed1-19ca4f88e98c.png)

---

The blurry effect can also be seen in this comment here – so it does not only happen to me:
https://github.com/endroid/qr-code/issues/348#issuecomment-920734030

**Comment by endroid** [comment]
Hi @sprain thank you. This should be fixed in 4.4.3.

**Comment by sprain** [comment]
Awesome! Thank you!
I did try to fix it but could not find the issue. Very glad you did 😊
